### PR TITLE
Dev/dmlf muddle

### PR DESCRIPTION
**Pull request by engineerkatie**
This adds the muddle comms system and a test.

### CODE REVIEW

File: libs/dmlf/tests/dmlf_muddle/dmlf_muddle.cpp [hunk 1]
```diff
@@ -0,0 +1,216 @@
+//------------------------------------------------------------------------------
```
Comment (JuanBesa) [inline libs/dmlf/tests/dmlf_muddle/dmlf_muddle.cpp:1]: We put the tests in tests/execution maybe this could also got there?

Comment (engineerkatie) [reply libs/dmlf/tests/dmlf_muddle/dmlf_muddle.cpp:1]: There's also an argument that they should go in tests/networking...

**Review by JuanBesa — DISMISSED** [review]
I don't really understand the muddle part, but it looks good to me.

File: libs/dmlf/include/dmlf/execution_workload.hpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    , state_name_(std::move(state_name))
+    , worker_(std::move(worker))
+  {}
+  virtual ~ExecutionWorkload() = default;
+
+  ExecutionWorkload(ExecutionWorkload const &other) = default;
+  ExecutionWorkload &operator=(ExecutionWorkload const &other) = default;
+
+  bool operator==(ExecutionWorkload const &other) = delete;
+  bool operator<(ExecutionWorkload const &other)  = delete;
```
Comment (JuanBesa) [inline libs/dmlf/include/dmlf/execution_workload.hpp:51]: These could be deleted

**Review by JuanBesa — DISMISSED** [review]

**Review by JuanBesa — APPROVED** [review]

**Pull request merged**